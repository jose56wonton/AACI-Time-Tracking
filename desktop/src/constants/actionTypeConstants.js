import { createCustomActionTypes,createCRUDActionTypes } from 'helpers/actionTypes';

const activityActionTypes = {
  ...createCRUDActionTypes('activity','activities') 
};


const analyzeActionTypes = {
  SELECT_EMPLOYEE: 'select_employee',
  SET_EMPLOYEE_STATUS: 'set_employee_status',  
  SELECT_TASK: 'select_task',
  SET_TASK_STATUS: 'set_task_status',  
  SELECT_PROJECT: 'select_project',
  SET_PROJECT_STATUS: 'set_project_status',  
  SELECT_SHIFT: 'select_shift',
  SET_SHIFT_STATUS: 'set_shift_status',  
  SELECT_CREW: 'select_crew',
  SET_CREW_STATUS: 'set_crew_status',  
  SELECT_AUTHORITY: 'select_authority',
  SET_AUTHORITY_STATUS: 'set_authority_status',  
  SELECT_CATEGORY: 'select_category',
  SET_CATEGORY_STATUS: 'set_category_status',  
  SELECT_SUBCATEGORY: 'select_subcategory',
  SET_SUBCATEGORY_STATUS: 'set_subcategory_status',
  SELECT: 'select_domain',
  SET_STATUS: 'set_domain_status',
  DELETE_SELECTED: 'delete_selected'
}; 

const authorityActionTypes = {
  ...createCRUDActionTypes('authority','authorities'),
  EDIT_AUTHORITIES_MODAL: 'edit_authorities_modal'
};

const categoryActionTypes = {
  ...createCRUDActionTypes('category','categories'),
  EDIT_CATEGORIES_MODAL: 'edit_categories_modal'
};

const crewActionTypes = {
  ...createCRUDActionTypes('crew','crews'),
  EDIT_CREWS_MODAL: 'edit_crews_modal'
};

const dimensionActionTypes = {
  ...createCRUDActionTypes('dimension','dimensions')
};
const employeeActionTypes = {
  ...createCRUDActionTypes('employee','employees'),
  ...createCustomActionTypes('employee','login'),
  ...createCustomActionTypes('employee','clockin'),
  CLOCKIN_EMPLOYEE_SNACK: 'clockin_employee_snack',
  ...createCustomActionTypes('employee','clockout'),
  CLOCKOUT_EMPLOYEE_SNACK: 'clockout_employee_snack'
};
const exportActionTypes = {
  ...createCustomActionTypes('excel','export')
};
const modalActionTypes = {
  HIDE_MODAL: 'hide_modal',
  SHOW_MODAL: 'show_modal'
};
const projectActionTypes = {
  ...createCRUDActionTypes('project','projects'),
  ...createCustomActionTypes('project','update'),
  ...createCustomActionTypes('project','create')
};

const projectTaskActionTypes = {
  ...createCRUDActionTypes('project_task','project_tasks')
};

const shiftActionTypes = {
  ...createCRUDActionTypes('shift','shifts'),
  ...createCustomActionTypes('shift','create'),
  ...createCustomActionTypes('shift','remove'),
  ...createCustomActionTypes('shift','edit'), 
  ...createCustomActionTypes('current_shift','get') ,  
  ...createCustomActionTypes('shifts_in_range','get') 
};
const snackActionTypes = {
  HIDE_SNACK: 'hide_snack',
  SHOW_SNACK: 'show_snack'
};

const staticActionTypes = {
  ...createCustomActionTypes('static_data','get')
};

const subcategoryActionTypes = {
  ...createCRUDActionTypes('subcategory','subcategories')
};
const taskActionTypes = {
  ...createCRUDActionTypes('task','tasks')
};

const userActionTypes = {
  ...createCustomActionTypes('user','login')
};


export {
  activityActionTypes,
  analyzeActionTypes,
  authorityActionTypes,
  categoryActionTypes,
  crewActionTypes,
  dimensionActionTypes,
  employeeActionTypes,
  exportActionTypes,
  modalActionTypes,
  projectActionTypes,
  projectTaskActionTypes,
  shiftActionTypes,
  snackActionTypes,
  staticActionTypes,
  subcategoryActionTypes,
  taskActionTypes,
  userActionTypes
};
