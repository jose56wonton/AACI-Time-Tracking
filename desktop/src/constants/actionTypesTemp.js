import { createCustomActionTypes,createCRUDActionTypes } from 'helpers/actionTypes';

const activityActionTypes = {
  ...createCRUDActionTypes('activity') 
};


const analyzeActionTypes = {
  SELECT_EMPLOYEE: 'select_employee',
  SET_EMPLOYEE_STATUS: 'set_employee_status',  
  SELECT_TASK: 'select_task',
  SET_TASK_STATUS: 'set_task_status',  
  SELECT_PROJECT: 'select_project',
  SET_PROJECT_STATUS: 'set_project_status',  
  SELECT_SHIFT: 'select_shift',
  SET_SHIFT_STATUS: 'set_shift_status',  
  SELECT_CREW: 'select_crew',
  SET_CREW_STATUS: 'set_crew_status',  
  SELECT_AUTHORITY: 'select_authority',
  SET_AUTHORITY_STATUS: 'set_authority_status',  
  SELECT_CATEGORY: 'select_category',
  SET_CATEGORY_STATUS: 'set_category_status',  
  SELECT_SUBCATEGORY: 'select_subcategory',
  SET_SUBCATEGORY_STATUS: 'set_subcategory_status'
}; 

const authorityActionTypes = {
  ...createCRUDActionTypes('authority'),
  EDIT_AUTHORITIES_MODAL: 'edit_authorities_modal'
};

const categoryActionTypes = {
  ...createCRUDActionTypes('category'),
  EDIT_CATEGORIES_MODAL: 'edit_categories_modal'
};

const crewActionTypes = {
  ...createCRUDActionTypes('crew'),
  EDIT_CREWS_MODAL: 'edit_crews_modal'
};

const dimensionActionTypes = {
  ...createCRUDActionTypes('dimension')
};
const employeeActionTypes = {
  ...createCRUDActionTypes('employee'),
  ...createCustomActionTypes('shift','login'),
  ...createCustomActionTypes('shift','clockin'),
  CLOCKIN_EMPLOYEE_SNACK: 'clockin_employee_snack',
  ...createCustomActionTypes('shift','clockout'),
  CLOCKOUT_EMPLOYEE_SNACK: 'clockout_employee_snack'
};
const exportActionTypes = {
  ...createCustomActionTypes('excel','export')
};
const modalActionTypes = {
  HIDE_MODAL: 'hide_modal',
  SHOW_MODAL: 'show_modal'
};
const projectActionTypes = {
  ...createCRUDActionTypes('project'),
  ...createCustomActionTypes('project','update'),
  ...createCustomActionTypes('project','create')
};

const projectTaskActionTypes = {
  ...createCRUDActionTypes('project_task')
};

const shiftActionTypes = {
  ...createCRUDActionTypes('shift'),
  ...createCustomActionTypes('shift','create'),
  ...createCustomActionTypes('shift','remove'),
  ...createCustomActionTypes('shift','edit'), 
  ...createCustomActionTypes('current_shift','get') ,  
  ...createCustomActionTypes('shifts_in_range','get') 
};
const snackActionTypes = {
  HIDE_SNACK: 'hide_snack',
  SHOW_SNACK: 'show_snack'
};

const staticActionTypes = {
  ...createCustomActionTypes('static_data','get')
};

const subcategoryActionTypes = {
  ...createCRUDActionTypes('subcategory')
};
const taskActionTypes = {
  ...createCRUDActionTypes('task')
};

const userActionTypes = {
  ...createCustomActionTypes('user','login')
};


export {
  activityActionTypes,
  analyzeActionTypes,
  authorityActionTypes,
  categoryActionTypes,
  crewActionTypes,
  dimensionActionTypes,
  employeeActionTypes,
  exportActionTypes,
  modalActionTypes,
  projectActionTypes,
  projectTaskActionTypes,
  shiftActionTypes,
  snackActionTypes,
  staticActionTypes,
  subcategoryActionTypes,
  taskActionTypes,
  userActionTypes
};
