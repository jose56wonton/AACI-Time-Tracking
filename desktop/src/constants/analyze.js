export const INIT = 'initial';
export const ADDING = 'adding';
export const EDITING = 'editing';