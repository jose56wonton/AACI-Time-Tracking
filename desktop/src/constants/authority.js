export const EMPLOYEE = `employee`;
export const ADMIN = `admin`;
export const MANAGER = `manager`;
