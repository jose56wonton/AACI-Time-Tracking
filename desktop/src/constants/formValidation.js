import * as Yup from 'yup';

export const auth = Yup.object().shape({
  username: Yup.string().required('Network is required'),
  password: Yup.string().required('Password is required')
});
export const account = Yup.object().shape({
  pin: Yup.string()
    .min(6, 'Pins are 6 characters')
    .max(6, 'Pins are 6 characters')
    .required('Pin required')
});
export const shift = Yup.object().shape({
  activities: Yup.array().of(
    Yup.object().shape({
      length: Yup.number()
        .min(1,'Time must be specified')
        .required('Time selection required'),
      projectId: Yup.number()
        .positive('Project selection required')
        .required('Project selection required'),
      projectTaskId: Yup.number()
        .positive('Task selection required')
        .required('Task selection required'),
      description: Yup.string()
    }),
  ).required('Activity selection required')
});
export const exportValidation = Yup.object().shape({
  exportCategory: Yup.number().min(0, 'Invalid export category'),
  start: Yup.date().required('Invalid start date'), 
  fileLocation: Yup.string()
    .required('File location is required')
    .matches(
      new RegExp(/\.(xls|xlsx)$/i),
      'File location must be an Xls or Xlsx file',
    )
});

export const employeeValidation = Yup.object().shape({
  firstName: Yup.string()
    .required('First Name is required')
    .matches(new RegExp(/[a-zA-Z]{2,}/),'First Name format is invalid'),
  lastName: Yup.string()
    .required('Last Name is required')
    .matches(new RegExp(/[a-zA-z]{1,}'?-?[a-zA-Z]{2,}\s?([a-zA-Z]{1,})?/), 'Last Name format is invalid'),
  authorityId: Yup.number()
    .positive('Authority selection is required')
    .required('Authority selection is required'),
  crewId: Yup.number()
    .positive('Crew selection is required')
    .required('Crew selection is required'),
  pin: Yup.string()
    .required('Pin is required')
    .matches(new RegExp(/\b(\d{6})\b/),'Pin format is invalid'),
  isEmployed: Yup.boolean(),
  isWorking: Yup.boolean()
});


