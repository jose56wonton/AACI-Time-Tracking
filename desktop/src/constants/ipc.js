exports.SET_CRED = `set_cred`;
exports.GET_CRED = `get_cred`;
exports.CREATE_EXPORT = `create_export`;

exports.TOGGLE_FULLSCREEN = `toggle_fullscreen`;
exports.IS_FULLSCREEN = `is_fullscreen`;