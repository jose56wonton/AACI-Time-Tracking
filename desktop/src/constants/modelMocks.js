
export const EMPLOYEE_MOCK = [{
  id: 0,
  firstName: '0',
  lastName: '0'
}, {
  id: 1,
  firstName: '1',
  lastName: '1'
}, {
  id: 2,
  firstName: '2',
  lastName: '2'
}]

export const PROJECT_MOCK = [{
  id: 0,
  name: '0',
  isActive: 0,
  date: "2017-02-13T00:02:00.000Z"
}, {
  id: 1,
  name: '1',
  isActive: 0,
  date: "2017-02-13T00:02:00.000Z"
}, {
  id: 2,
  name: '2',
  isActive: 0,
  date: "2017-02-13T00:02:00.000Z"
}]

export const PROJECT_TASK_MOCK = [{
  id: 0,
  estimateTime: 758,
  projectId: 0,
  quantity: 1500,
  taskId: 0,
  task: {
    id: 0,
    isActive: 0,
    name: "0",
    category: { id: 0, name: 'name0' },
    subcategory: { id: 0, name: 'name0' }
  }
},
{
  id: 1,
  estimateTime: 758,
  projectId: 1,
  quantity: 1500,
  taskId: 1,
  task: {
    id: 1,
    isActive: 1,
    name: "1",
    category: { id: 1, name: 'name1' },
    subcategory: { id: 1, name: 'name1' }
  }
},
{
  id: 2,
  estimateTime: 758,
  projectId: 2,
  quantity: 1500,
  taskId: 2,
  task: {
    id: 2,
    isActive: 2,
    name: "2",
    category: { id: 2, name: 'name2' },
    subcategory: { id: 2, name: 'name2' }
  }
}]

export const ACTIVITY_MOCK = [{
  description: "",
  id: 1,
  length: 225,
  projectTaskId: 1,
  shiftId: 1,
  projectTask: {
    estimateTime: 758,
    id: 1,
    projectId: 1,
    quantity: 1500,
    taskId: 1,
    task: {
      id: 1,
      isActive: 1,
      name: "6\" Driveway",
      subcategoryId: 1,
      subcategory: {
        categoryId: 1,
        id: 1,
        type: "Subcategory Name",
        dimensionId: 1
      },
      category: {
        id: 1,
        type: "Category Name"
      },
      dimension: {
        id: 1,
        type: "Dimension Name"
      }
    },
    project: {
      date: "2017-02-13T00:02:00.000Z",
      id: 1,
      isActive: 0,
      name: "1722: Musc. Co. Fruitland Whitetopping"
    }
  },
  projectId: 1
}]