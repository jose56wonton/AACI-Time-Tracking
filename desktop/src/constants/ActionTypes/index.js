import * as userActionTypes from './user';
import shiftActionTypes from './shift';
import * as snackActionTypes from './snack';
import * as employeeActionTypes from './employee';
import * as crewActionTypes from './crew';
import * as authorityActionTypes from './authority';
import * as projectActionTypes from './project';
import * as projectTaskActionTypes from './projectTask';
import * as taskActionTypes from './task';
import * as activityActionTypes from './activity';
import * as staticActionTypes from './static';
import * as exportActionTypes from './export';
import * as analyzeActionTypes from './analyze';
import * as dimensionActionTypes from './dimension';
import * as categoryActionTypes from './category';
import * as subcategoryActionTypes from './subcategory';
import * as modalActionTypes from './modal';

export {
  userActionTypes,
  shiftActionTypes,
  snackActionTypes,
  employeeActionTypes,
  crewActionTypes,
  authorityActionTypes,
  projectActionTypes,
  projectTaskActionTypes,
  taskActionTypes,
  activityActionTypes,
  staticActionTypes,
  exportActionTypes,
  analyzeActionTypes,
  dimensionActionTypes,
  categoryActionTypes,
  subcategoryActionTypes,
  modalActionTypes
};
