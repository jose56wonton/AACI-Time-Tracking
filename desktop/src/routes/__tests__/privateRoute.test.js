import React from 'react';
import { shallow } from 'enzyme';

import PrivateRouteHOC, { PrivateRoute } from 'routes/privateRoute';
import Hero from 'components/layouts/Hero';
import * as status from 'constants/status';

const props = {
  component: Hero,
  user: {
    status: status.SUCCESS
  },
  location: {}
};
const setup = overRides => {  
  return shallow(<PrivateRoute {...props} {...overRides}/>);
};

const setupHOC = overRides => {
  return shallow(<PrivateRouteHOC {...props} {...overRides} />);
};

describe('Private Route', () => {
  it('should render without error', () => {
    const wrapper = setup();
    const Render = wrapper.prop('render');
    shallow(<Render />)
    expect(wrapper).toMatchSnapshot();
  });
  it('should redirect if user.state != status.SUCCESS', () => {
    const wrapper = setup({user: {status: status.FAILURE}});
    const Render = wrapper.prop('render');
    shallow(<Render />)
    expect(wrapper).toMatchSnapshot();
  })
  // it('should render correctly with connect wrapper', () => {
  //   const wrapper = setupHOC();
  // });
});