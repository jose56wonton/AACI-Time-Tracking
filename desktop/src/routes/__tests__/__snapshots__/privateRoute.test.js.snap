// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private Route should redirect if user.state != status.SUCCESS 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`Private Route should render without error 1`] = `
<Route
  render={[Function]}
/>
`;
